modules/mbedtls_curl/http_client_curl.cpp

```diff
@@ -0,0 +1,471 @@
+			break;
+		case METHOD_DELETE:
+			break;
+		case METHOD_OPTIONS:
+			break;
+		case METHOD_TRACE:
+			break;
+		case METHOD_CONNECT:
+			break;
+		case METHOD_PATCH:
+			break;
+		case METHOD_MAX:
+			break;
+	}
+
+	// Somewhat counter intuitively, the read function is actually used by libcurl to send data,
+	// while the write function (see below) is used by libcurl to write response data to storage
+	// (or in our case, memory).
+	curl_easy_setopt(p_chandle, CURLOPT_READFUNCTION, _read_callback);
+	curl_easy_setopt(p_chandle, CURLOPT_READDATA, &b);
+	read_buffer = b;
+}
+
+Error HTTPClientCurl::_init_dns(CURL *p_chandle, IPAddress p_addr) {
+	// TODO: Support resolving multiple addresses.
+	curl_slist *h = _ip_addr_to_slist(p_addr);
+	CURLcode return_code = curl_easy_setopt(p_chandle, CURLOPT_RESOLVE, h);
+	if (return_code != CURLE_OK) {
+		ERR_PRINT("failed to initialize dns resolver: " + String::num_int64(return_code));
+		return FAILED;
+	}
+	return OK;
+}
+
+Error HTTPClientCurl::_init_request_headers(CURL *p_chandler, Vector<String> p_headers) {
+	curl_slist *h = nullptr;
+	for (int i = 0; i < p_headers.size(); i++) {
+		h = curl_slist_append(h, p_headers[i].ascii().get_data());
+	}
+	if (h) {
+		CURLcode return_code = curl_easy_setopt(p_chandler, CURLOPT_HTTPHEADER, h);
+		if (return_code != CURLE_OK) {
+			ERR_PRINT("failed to set request headers: " + String::num_uint64(return_code));
+			return FAILED;
+		}
+	}
+	return OK;
+}
+
+Error HTTPClientCurl::_request(bool p_init_dns) {
+	String h = host;
+	if (h.is_valid_ip_address() && h.find(":") != -1) {
+		h = "[" + h + "]";
+	}
+
+	CURL *eh = curl_easy_init();
+	curl_easy_setopt(eh, CURLOPT_URL, (scheme + h + ":" + String::num_int64(port) + url).ascii().get_data());
+	curl_easy_setopt(eh, CURLOPT_CUSTOMREQUEST, methods[(int)method]);
+	curl_easy_setopt(eh, CURLOPT_BUFFERSIZE, read_chunk_size);
+	Error err;
+	if (p_init_dns) {
+		err = _init_dns(eh, addr);
+		if (err != OK) {
+			return err;
+		}
+	}
+
+	if (request_body_size > 0) {
+		_init_upload(eh, method, (uint8_t *)request_body, request_body_size);
+	}
+
+	if (ssl) {
+		curl_easy_setopt(eh, CURLOPT_USE_SSL, CURLUSESSL_ALL);
+
+		if (ca_path != "") {
+			curl_easy_setopt(eh, CURLOPT_CAINFO, ca_path.ascii().get_data());
+		} else {
+			curl_blob ca_blob;
+			ca_blob.data = (uint8_t *)ca_data.ptr();
+			ca_blob.len = ca_data.size();
+			ca_blob.flags = CURL_BLOB_COPY;
+
+			curl_easy_setopt(eh, CURLOPT_CAINFO_BLOB, &ca_blob);
+		}
+	}
+
+	if (!verify_host) {
+		// When CURLOPT_SSL_VERIFYHOST is 2 (the default), that certificate must indicate
+		// that the server is the server to which you meant to connect, or
+		// the connection fails. Simply put, it means it has to have the same
+		// name in the certificate as is in the URL you operate against.
+		// @see https://curl.se/libcurl/c/CURLOPT_SSL_VERIFYHOST.html
+		curl_easy_setopt(eh, CURLOPT_SSL_VERIFYHOST, 0L);
```

Comment:
Was this supposed to be 2?

Reply:
The diff was confusing. This is correct to not verify host. 

Can you rebase to like 2-3 commits?

Reply:
Yep. Ive actually already done so locally. I was just waiting for CI to finish before I pushed. Ill do so shortly.